Doc/library/ctypes.rst

```diff
@@ -1408,6 +1408,10 @@ accessing the function through an attribute caches the result and therefore
 accessing it repeatedly returns the same object each time.  On the other hand,
 accessing it through an index returns a new object each time:
 
+.. doctest::
```

Comment:
I find your version much clearer, +1. However, I don't see much point converting this to a doctest. I'd say remove `.. doctest::` and keep the rest of the changes.